ddtrace/contrib/anthropic/_streaming.py

```diff
@@ -0,0 +1,322 @@
+    if getattr(chunk, "type", "") != "content_block_delta":
+        return message
+
+    delta_block = getattr(chunk, "delta", "")
+    chunk_content = getattr(delta_block, "text", "")
+    if chunk_content:
+        message["content"][-1]["text"] += chunk_content
+    return message
+
+
+def _on_message_delta_chunk(chunk, message):
+    # message delta events signal the end of the message
+    if getattr(chunk, "type", "") != "message_delta":
+        return message
+
+    delta_block = getattr(chunk, "delta", "")
+    chunk_finish_reason = getattr(delta_block, "stop_reason", "")
+    if chunk_finish_reason:
+        message["finish_reason"] = chunk_finish_reason
+        message["content"][-1]["text"] = message["content"][-1]["text"].strip()
+
+    chunk_usage = getattr(chunk, "usage", {})
+    if chunk_usage:
+        message_usage = message.get("usage", {"output_tokens": 0, "input_tokens": 0})
+        message_usage["output_tokens"] += getattr(chunk_usage, "output_tokens", 0)
+        message_usage["input_tokens"] += getattr(chunk_usage, "input_tokens", 0)
+        message["usage"] = message_usage
+
+    return message
+
+
+# To-Do: Handle error blocks appropriately
+# def _on_error_chunk(chunk, message):
+#     # this is the start to a message.content block (possibly 1 of several content blocks)
+#     if getattr(chunk, "type", "") != "error":
+#         return message
+
+#     message["content"].append({"type": "text", "text": ""})
+#     return message
+
+
+def _tag_streamed_chat_completion_response(integration, span, message):
+    """Tagging logic for streamed chat completions."""
+    if message is None:
+        return
+    for idx, block in enumerate(message["content"]):
+        span.set_tag_str("anthropic.response.completions.content.%d.type" % idx, str(integration.trunc(block["type"])))
+        span.set_tag_str("anthropic.response.completions.content.%d.text" % idx, str(integration.trunc(block["text"])))
+        span.set_tag_str("anthropic.response.completions.role", str(message["role"]))
+        if message.get("finish_reason") is not None:
+            span.set_tag_str("anthropic.response.completions.finish_reason", str(message["finish_reason"]))
+
+    usage = _get_attr(message, "usage", {})
+    integration.record_usage(span, usage)
+
+
+def _is_stream(resp):
+    # type: (...) -> bool
+    import anthropic
+
+    if hasattr(anthropic, "Stream") and isinstance(resp, anthropic.Stream):
+        return True
+    return False
+
+
+def _is_async_stream(resp):
+    # type: (...) -> bool
+    import anthropic
+
+    if hasattr(anthropic, "AsyncStream") and isinstance(resp, anthropic.AsyncStream):
+        return True
+    return False
+
+
+def _is_stream_manager(resp):
+    # type: (...) -> bool
+    import anthropic
+
+    if hasattr(anthropic, "MessageStreamManager") and isinstance(resp, anthropic.MessageStreamManager):
+        return True
+    return False
+
+
+def _is_async_stream_manager(resp):
+    # type: (...) -> bool
```

Comment:
```suggestion
def _is_async_stream_manager(resp: Any) -> bool:
```

same for the others.